lib/utils/sudo.ts

```diff
@@ -14,34 +14,78 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { spawn, StdioOptions } from 'child_process';
 
 import * as Bluebird from 'bluebird';
-import * as rindle from 'rindle';
+import { ChildProcess, spawn, SpawnOptions } from 'child_process';
+import { stripIndent } from 'common-tags';
+import * as _ from 'lodash';
 
+/**
+ * Execute a child process with admin / superuser privileges, prompting
+ * the user for elevation as needed.
+ * @param command Path to an executable, plus any arguments.
+ * These will be shell-escaped by this function.
+ * @param stderr Optional stream to which stderr should be piped
+ */
 export async function executeWithPrivileges(
 	command: string[],
 	stderr?: NodeJS.WritableStream,
-): Promise<string> {
-	const stdio: StdioOptions = [
-		'inherit',
-		'inherit',
-		stderr ? 'pipe' : 'inherit',
-	];
-	const opts = {
-		env: process.env,
-		stdio,
-	};
+): Promise<void> {
+	// whether the CLI is already running with admin / super user privileges
+	const isElevated = await (await import('is-elevated'))();
+	const { shellEscape } = await import('./helpers');
+	const escapedArgs = shellEscape(command);
 
-	const args = process.argv
-		.slice(0, 2)
-		.concat(['internal', 'sudo', command.join(' ')]);
+	await new Bluebird((resolve, reject) => {
+		const opts: SpawnOptions = {
+			env: process.env,
+			shell: true,
+			stdio: ['inherit', 'inherit', stderr ? 'pipe' : 'inherit'],
+		};
+		let ps: ChildProcess;
+		let errMsgCmd: string;
 
-	const ps = spawn(args[0], args.slice(1), opts);
+		if (isElevated) {
+			// already running with privileges: simply spawn the command
+			errMsgCmd = escapedArgs.join(' ');
+			ps = spawn(escapedArgs[0], escapedArgs.slice(1), opts);
```

Comment:
`if (isElevated)` addresses the following cases:
* A small performance improvement by avoiding calling sudo or windosu when the CLI is already being executed a root / admin. In this case, the Windows / Linux / Mac code paths all run this same `spawn` line (as opposed to `windosu` vs `sudo`).
* Some Docker containers in CI environments may not even have `sudo` installed -- so it's one fewer dependency to worry about. It's actually often how I test the CLI on Linux on a balenaOS NUC computer, using a minimalistic Debian container that doesn't come with sudo unless I `apt-get` it.